examples/jsm/nodes/procedural/Fractal3DNode.js

```diff
@@ -0,0 +1,110 @@
+import { TempNode } from '../core/TempNode.js';
+import { FunctionNode } from '../core/FunctionNode.js';
+import { IntNode } from '../inputs/IntNode.js';
+import { FloatNode } from '../inputs/FloatNode.js';
+import { PositionNode } from '../accessors/PositionNode.js';
+import { Noise2DNode } from './Noise2DNode.js';
+import { Noise3DNode } from './Noise3DNode.js';
+
+const FRACTAL3D_SRC = `
+float fractal3d( vec3 p, float amplitude, int octaves, float lacunarity, float diminish ) {
+
+	float result = 0.0;
+
+	for (int i = 0;  i < octaves; ++i) {
+
+		result += amplitude * noise3d(p);
+		amplitude *= diminish;
+		p *= lacunarity;
+
+	}
+
+	return result;
+
+}
+`.trim();
+
+/** Fractional Brownian motion. */
+function Fractal3DNode( position = new PositionNode(), amplitude = new FloatNode( 1.0 ), octaves = 3.0, lacunarity = 2.0, diminish = 0.5 ) {
+
+	TempNode.call( this, 'f' );
+
+	this.position = position;
+	this.amplitude = amplitude;
+	this.octaves = new IntNode( octaves ).setReadonly( true );
+	this.lacunarity = new FloatNode( lacunarity ).setReadonly( true );
+	this.diminish = new FloatNode( diminish ).setReadonly( true );
+
+}
+
+Fractal3DNode.prototype = Object.create( TempNode.prototype );
+Fractal3DNode.prototype.constructor = Fractal3DNode;
+Fractal3DNode.prototype.nodeType = 'Fractal3D';
+
+Fractal3DNode.Nodes = (function () {
+
+	return {
+
+		fractal3d: new FunctionNode( FRACTAL3D_SRC )
+
+	};
+
+})();
+
+Fractal3DNode.prototype.generate = function (builder, output) {
+
+	const noiseCommon = builder.include( Noise2DNode.Nodes.noiseCommon );
+	const noise3D = builder.include( Noise3DNode.Nodes.noise3d );
+	const fractal3d = builder.include( Fractal3DNode.Nodes.fractal3d );
+
+	return builder.format( fractal3d + '( ' + [
+
+		this.position.build( builder ),
```

Comment:
It would be nice to include the input type like: `.build( builder, 'vec3' )` to allow automatic conversion. Although I recommend that you migrate of `builder.include` to `new FunctionNodeCall( Noise3DNode.Nodes.fractal3d )` and add the includes depedencies in own `FunctionNode`.

Reply:
I think I've fixed most of this but I'm not sure what "add the includes dependencies in own `FunctionNode`" means, is there an example of that somewhere?

Reply:
You can use thus: 
https://github.com/mrdoob/three.js/blob/35eaf1906fb74f16be8b155c86c1e0d8a1d4669c/examples/webgl_materials_nodes.html#L1390

or like this
`myFunctionNode.includes = [ myFunctionNodeDepedencies ]`

https://github.com/mrdoob/three.js/blob/35eaf1906fb74f16be8b155c86c1e0d8a1d4669c/examples/webgl_materials_nodes.html#L2682

Reply:
Done, thanks!